pkg/gateway/operations/deleteobject.go

```diff
@@ -26,11 +26,24 @@ func (controller *DeleteObject) HandleAbortMultipartUpload(w http.ResponseWriter
 	o.Incr("abort_mpu", o.Principal, o.Repository.Name, o.Reference)
 	query := req.URL.Query()
 	uploadID := query.Get(QueryParamUploadID)
+
+	mpu, err := o.MultipartTracker.Get(req.Context(), uploadID)
+	if err != nil {
+		o.Log(req).WithError(err).Error("upload id not found in tracker")
+		_ = o.EncodeError(w, req, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrNoSuchKey))
+		return
+	}
+	if mpu.Path != o.Path {
```

Comment:
Is this really possible?

Reply:
yes, s3 fails in case the upload doesn't match the key in the request